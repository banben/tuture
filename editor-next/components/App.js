import { Row, Col } from 'antd';

import Content from './Content';
import StepFileList from './StepFileList';

function App() {
  return (
    <div>
      <Row>
        <Col xs={0} md={0} lg={5}>
          <div className="assistLayout">辅助布局填充</div>
        </Col>
        <Col md={24} lg={19} xl={14}>
          <div className="content">
            <Content />
          </div>
        </Col>
        <Col xs={0} md={0} xl={5}>
          <div className="codeFileList">
            <StepFileList />
          </div>
        </Col>
      </Row>
    </div>
  );
}

export default App;
