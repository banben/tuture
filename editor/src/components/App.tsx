import React from 'react';
import styled, { injectGlobal } from 'styled-components';
import { inject, observer } from 'mobx-react';
import { translate } from 'react-i18next';
import { Helmet } from 'react-helmet';

import SideBarLeft from './SideBarLeft';
import SideBarRight from './SideBarRight';
import { DiffItem } from './DiffView';
import Content from './Content';
import { Tuture } from '../../../types';
import { extractCommits, handleAnchor, vwDesign, vwFontsize } from '../utils';
import Header from './Header';
import Store from '../store';
import NoCommit from './NoCommit.png';

export interface AppProps {
  tuture?: Tuture | string;
  diff?: DiffItem[] | string;
  store?: Store;
  i18n?: any;
}

interface AppState extends AppProps {
  hasCommit?: boolean;
}

const AppContent = styled.div`
  width: 86%;

  @media (max-width: 1408px) {
    width: 90%;
  }

  @media (max-width: 1206px) {
    width: 94%;
  }

  margin: 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-top: 60px;
`;

const NoCommitContainer = styled.div`
  height: 100vh;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 200px;
`;

const NoCommitLogo = styled.img`
  height: 300px;
`;

const NoCommitText = styled.p`
  margin-top: 40px;
  font-size: 20px;
`;

injectGlobal`
  html {
    font-size: ${(vwFontsize / vwDesign) * 100}vw;
    min-width: 1080px;
  }

  body {
    height: 100%;
    font-size: 14px;
    line-height: 1.5;
    margin: 0;
    font-family: Roboto;
  }

  #root {
    height: 100%;
    margin-top: 10px;
    margin-bottom: 70px;
  }

  h1 {
    font-size: 45px;
  }

  .showSideBar {
    opacity: 1;
  }

  .hideSideBar {
    opacity: 0;
  }
`;

export const ModeContext = React.createContext({
  toggleEditMode: () => {},
});

@inject('store')
@observer
class App extends React.Component<AppProps, AppState> {
  constructor(props: AppProps) {
    super(props);

    const { tuture } = props;

    // Judge whether project has commit
    if (
      tuture &&
      (tuture as Tuture).steps &&
      Array.isArray((tuture as Tuture).steps) &&
      (tuture as Tuture).steps.length > 0
    ) {
      const { diff, i18n, store } = props;
      const nowAnchorName = (tuture as Tuture).steps[0].name;
      store.setTuture(tuture as Tuture);
      store.nowSelected = handleAnchor(nowAnchorName);
      store.i18n = i18n;

      this.state = {
        diff,
        hasCommit: true,
      };
    } else {
      this.state = {
        hasCommit: false,
      };
    }
  }

  toggleEditMode = () => {
    const { store } = this.props;
    store.updateIsEditMode();
    if (!store.isEditMode) {
      store.saveTuture();
    }
  };

  render() {
    let bodyContent: React.ReactNode;
    let tutorialTitle = '图雀';

    const hasCommit = false;

    if (hasCommit) {
      const { diff } = this.state;
      const { store } = this.props;
      const { tuture } = store;

      if (tuture && tuture.name) {
        tutorialTitle = tuture.name;
      }

      const commits = extractCommits(tuture as Tuture);
      bodyContent = [
        <SideBarLeft commits={commits} key="SiderBarLeft" />,
        <Content diff={diff} key="Content" />,
        this.props.store.isEditMode && <SideBarRight key="SideBarRight" />,
      ];
    } else {
      bodyContent = (
        <NoCommitContainer>
          <NoCommitLogo src={NoCommit} alt="" />
          <NoCommitText>Oops！ 此项目还没有任何 commit 哦！</NoCommitText>
        </NoCommitContainer>
      );
    }

    return (
      <ModeContext.Provider value={{ toggleEditMode: this.toggleEditMode }}>
        <Helmet>
          <title>{tutorialTitle}</title>
        </Helmet>
        {hasCommit && <Header />}
        {hasCommit && <AppContent>{bodyContent}</AppContent>}
        {!hasCommit && bodyContent}
      </ModeContext.Provider>
    );
  }
}

export default translate('translations')(App);
