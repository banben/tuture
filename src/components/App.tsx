import React from 'react';
import styled, { injectGlobal } from 'styled-components';
import fetch from 'isomorphic-fetch';
import { inject, observer } from 'mobx-react';

import SideBarLeft from './SideBarLeft';
import SideBarRight from './SideBarRight';
import { DiffItem } from './DiffView';
import Content from './Content';
import { Tuture } from '../types/';
import { extractCommits } from '../utils/extractors';
import Header from './Header';
import { handleAnchor, vwDesign, vwFontsize } from '../utils/common';
import Store from './store';

export interface AppProps {
  tuture?: Tuture | string;
  diff?: DiffItem[] | string;
  store?: Store;
}

interface AppState extends AppProps {}

const AppContent = styled.div`
  width: 86%;

  @media (max-width: 1408px) {
    width: 90%;
  }

  @media (max-width: 1206px) {
    width: 94%;
  }

  margin: 0 auto;
  display: flex;
  justify-content: space-between;
  margin-top: 60px;
`;

injectGlobal`
  html {
    font-size: ${(vwFontsize / vwDesign) * 100}vw;
  }

  body {
    height: 100%;
    font-size: 14px;
    line-height: 1.5;
    margin: 0;
  }

  #root {
    height: 100%;
    margin-top: 10px;
    margin-bottom: 70px;
  }

  h1 {
    font-size: 45px;
  }
`;

export const ModeContext = React.createContext({
  toggleEditMode: () => {},
});

@inject('store')
@observer
export default class App extends React.Component<AppProps, AppState> {
  constructor(props: AppProps) {
    super(props);

    let { tuture, diff } = props;
    const { store } = props;
    tuture = JSON.parse(tuture as string);
    diff = JSON.parse(diff as string);
    const nowAnchorName = (tuture as Tuture).steps[0].name;
    store.setTuture(tuture as Tuture);
    store.nowSelected = handleAnchor(nowAnchorName);

    this.state = {
      diff,
    };
  }

  saveTuture = () => {
    fetch(`http://${location.host}/save`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
      body: JSON.stringify(this.props.store.tuture),
    });
  };

  toggleEditMode = () => {
    const { store } = this.props;
    store.updateIsEditMode();
    if (!store.isEditMode) {
      this.saveTuture();
    }
  };

  render() {
    let bodyContent: React.ReactNode;

    const { diff } = this.state;
    const { tuture } = this.props.store;
    if (
      !tuture ||
      Object.keys(tuture).length === 0 ||
      !diff ||
      !Array.isArray(diff)
    ) {
      bodyContent = null;
    } else {
      const commits = extractCommits(tuture as Tuture);
      bodyContent = [
        <SideBarLeft commits={commits} key="SiderBarLeft" />,
        <Content diff={diff} key="Content" />,
        this.props.store.isEditMode && <SideBarRight key="SideBarRight" />,
      ];
    }

    return (
      <ModeContext.Provider
        value={{
          toggleEditMode: this.toggleEditMode,
        }}>
        <Header />
        <AppContent>{bodyContent}</AppContent>
      </ModeContext.Provider>
    );
  }
}
