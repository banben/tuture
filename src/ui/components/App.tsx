import React from 'react';
import styled, { injectGlobal } from 'styled-components';
import { inject, observer } from 'mobx-react';
import { translate } from 'react-i18next';
import { Helmet } from 'react-helmet';

import SideBarLeft from './SideBarLeft';
import SideBarRight from './SideBarRight';
import { DiffItem } from './DiffView';
import Content from './Content';
import { extractCommits, handleAnchor, vwDesign, vwFontsize } from '../utils';
import Header from './Header';
import Store from '../store';

export interface AppProps {
  tuture?: Tuture | string;
  diff?: DiffItem[] | string;
  store?: Store;
  i18n?: any;
}

interface AppState extends AppProps {}

const AppContent = styled.div`
  width: 86%;

  @media (max-width: 1408px) {
    width: 90%;
  }

  @media (max-width: 1206px) {
    width: 94%;
  }

  margin: 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-top: 60px;
`;

injectGlobal`
  html {
    font-size: ${(vwFontsize / vwDesign) * 100}vw;
    min-width: 1080px;
  }

  body {
    height: 100%;
    font-size: 14px;
    line-height: 1.5;
    margin: 0;
  }

  #root {
    height: 100%;
    margin-top: 10px;
    margin-bottom: 70px;
  }

  h1 {
    font-size: 45px;
  }

  .showSideBar {
    opacity: 1;
  }

  .hideSideBar {
    opacity: 0;
  }
`;

export const ModeContext = React.createContext({
  toggleEditMode: () => {},
});

@inject('store')
@observer
class App extends React.Component<AppProps, AppState> {
  constructor(props: AppProps) {
    super(props);

    const { tuture, diff, i18n, store } = props;
    const nowAnchorName = (tuture as Tuture).steps[0].name;
    store.setTuture(tuture as Tuture);
    store.nowSelected = handleAnchor(nowAnchorName);
    store.i18n = i18n;

    this.state = {
      diff,
    };
  }

  toggleEditMode = () => {
    const { store } = this.props;
    store.updateIsEditMode();
    if (!store.isEditMode) {
      store.saveTuture();
    }
  };

  render() {
    let bodyContent: React.ReactNode;
    let tutorialTitle = '图雀';

    const { diff } = this.state;
    const { store } = this.props;
    const { tuture } = store;

    if (tuture && tuture.name) {
      tutorialTitle = tuture.name;
    }

    if (
      !tuture ||
      Object.keys(tuture).length === 0 ||
      !diff ||
      !Array.isArray(diff)
    ) {
      bodyContent = null;
    } else {
      const commits = extractCommits(tuture as Tuture);
      bodyContent = [
        <SideBarLeft commits={commits} key="SiderBarLeft" />,
        <Content diff={diff} key="Content" />,
        this.props.store.isEditMode && <SideBarRight key="SideBarRight" />,
      ];
    }

    return (
      <ModeContext.Provider value={{ toggleEditMode: this.toggleEditMode }}>
        <Helmet>
          <title>{tutorialTitle}</title>
        </Helmet>
        <Header />
        <AppContent>{bodyContent}</AppContent>
      </ModeContext.Provider>
    );
  }
}

export default translate('translations')(App);
