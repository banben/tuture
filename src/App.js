import React from 'react';
import { Helmet } from 'react-helmet';
import yaml from 'js-yaml';
// TODO: hackable method to solve tuture.yml update not update browser

import './App.css';
import tuturePath from './path.json';

// import conponents
import Content from './components/Content';
import Catalog from './components/Catalog';

class App extends React.Component {
  state = {
    selectKey: '0',
    tuture: null,
    viewType: 'unified',
  };

  changeViewType = () => {
    const { viewType } = this.state;

    this.setState({
      viewType: viewType === 'unified' ? 'split' : 'unified',
    });
  }

  updateSelect = (key) => {
    this.setState({
      selectKey: key,
    });
  }

  async loadTuture() {
    const that = this;

    // Use frontend get data method to get tuture.yml
    const response = await fetch(`./tuture.yml`);
    const content = await response.text();

    // use js-yaml read yamm as js object
    const tuture = yaml.safeLoad(content);

    that.setState({
      tuture,
    });
  }

  componentDidMount() {
    this.loadTuture();
  }

  render() {
    const { tuture } = this.state;
    if (!tuture) {
      return null;
    }

    const catalogs = tuture.steps.map(item => ({
      name: item.name,
      commit: item.commit,
    }));
    const { name, language, maintainer, topics } = tuture;
    const catalogsInfo = {
      name,
      language,
      maintainer,
      topics,
    };
    const nowSelectKeyNumber = Number(this.state.selectKey);
    const nowRenderContent = tuture.steps[nowSelectKeyNumber];

    return (
      <div style={{ height: '100%', width: '100%', display: 'flex' }}>
        <Helmet>
          <title>{name}</title>
        </Helmet>
        <Catalog
          catalogs={catalogs}
          catalogsInfo={catalogsInfo}
          selectKey={this.state.selectKey}
          updateSelect={this.updateSelect}
        />
        <Content
          content={nowRenderContent}
          viewType={this.state.viewType}
          changeViewType={this.changeViewType}
        />
      </div>
    );
  }
}

export default App;
