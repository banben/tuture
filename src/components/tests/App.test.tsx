import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

test('Test App render successfully', () => {
  const component = renderer.create(
    <App />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
