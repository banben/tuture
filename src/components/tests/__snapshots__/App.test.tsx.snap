// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App render successfully 1`] = `
<div
  className="sc-bZQynM bmKOqU"
/>
`;
